src/ToDo/Business/TaskService.cs

```diff
@@ -40,4 +40,31 @@ public async Task DeleteAsync(ToDoTask task, CancellationToken ct)
 
 		_messenger.Send(new EntityMessage<ToDoTask>(EntityChange.Delete, task), task.ListId);
 	}
+
+	/// <inheritdoc />
+	public async ValueTask<IImmutableList<ToDoTask>> GetAllAsync(string displayName = "", CancellationToken ct = default)
+	{
+		if(String.IsNullOrWhiteSpace(displayName))
+		{
+			return ((await _client.GetAllAsync(ct)).Value ?? Enumerable.Empty<TaskData>())
```

Comment:
Extract GetAllAsync and GetByFilterAsync out as a variable and then condense the rest of the logic instead of duplicating it

Reply:
Sure